symbolicationWebService.py

```diff
@@ -146,50 +167,57 @@ def ReadConfigFile():
+    logging.error("Usage: symbolicationWebService.py [<config file>]")
     return False
   elif len(sys.argv) == 2:
     try:
-      configParser = ConfigParser.ConfigParser()
-      # Make parser case-sensitive
-      configParser.optionxform=str
+      # ConfigParser uses the pattern %(<variable>)<type> for variable substitution,
+      # so '%' found in environment variable values will raise an error. We replace
+      # '%' by '%%' to make the parser understand it is literal character.
+      environ = {key:value.replace(r"%", r"%%") for key, value in os.environ.iteritems()}
+
+      configParser = CaseSensitiveConfigParser(environ)
       configFile = open(sys.argv[1], "r")
       configParser.readfp(configFile)
       configFile.close()
     except ConfigParser.Error as e:
-      LogError("Unable to parse config file " + sys.argv[1] + ": " + str(e))
-    except:
-      LogError("Unable to open config file " + sys.argv[1])
+      logging.error("Unable to parse config file %s: %s", sys.argv[1], e)
+    except Exception as e:
+      logging.error("Unable to open config file %s: %s", sys.argv[1], e)
       return False
 
   # Check for section names
-  if set(configParser.sections()) != set(["General", "SymbolPaths", "SymbolURLs"]):
-    LogError("Config file should be made up of three sections: 'General', 'SymbolPaths' and 'SymbolURLs'")
+  if not set(["General", "Log"]).issubset(set(configParser.sections())):
+    logging.error("'General' and 'Log' sections are mandatory in the config file")
     return False
 
   generalSectionOptions = configParser.items("General")
   for (option, value) in generalSectionOptions:
     if option not in gOptions:
-      LogError("Unknown config option '" + option + "' in the 'General' section of config file")
+      logging.error("Unknown config option '" + option + "' in the 'General' section of config file")
       return False
     elif type(gOptions[option]) == int:
       try:
         value = int(value)
       except ValueError:
-        LogError("Integer value expected for config option '" + option + "'")
+        logging.error("Integer value expected for config option '" + option + "'")
         return False
     gOptions[option] = value
 
   # Get the list of symbol paths from the config file
-  configPaths = configParser.items("SymbolPaths")
-  if configPaths:
-    # Drop defaults if config file entries exist
-    gOptions["symbolPaths"] = [path for name, path in configPaths]
+  if configParser.has_section("SymbolPaths"):
+    configPaths = configParser.items("SymbolPaths")
+    if configPaths:
+      # Drop defaults if config file entries exist
+      gOptions["symbolPaths"] = [path for name, path in configPaths]
 
   # Get the list of symbol URLs from the config file
-  configURLs = configParser.items("SymbolURLs")
-  if configURLs:
-    gOptions["symbolURLs"] = [url for name, url in configURLs]
+  if configParser.has_section("SymbolURLs"):
```

Comment:
so you're going to allow having both SymbolPaths and SymbolURLs? yeah I guess that could be a valid config